core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -839,12 +863,22 @@
         .build();
     final String relJson = RelOptUtil.dumpPlan("", rel,
         SqlExplainFormat.JSON, SqlExplainLevel.EXPPLAN_ATTRIBUTES);
-    final String result = deserializeAndDumpToTextFormat(getSchema(rel), relJson);
-    final String expected = ""
+    final String result = deserializeAndDumpToTextFormat(getSchema(rel), relJson, format);
+    assertThat(result, isLinux(expected));
+  }
+
+  @Test void testUDAF() {
+    testUDAF0(SqlExplainFormat.TEXT, ""
```

Comment:
Can we make `RelWriterTest` parameterized instead of modifying and abstract common test base ?

Reply:
Sounds reasonable. 
I was having similar thoughts, but gave up because
1. some test cases in the class are not suitable for parameterized testing.
2. since the strings of the expected results take up much space of the class, parameterizing the class does not significantly reduce the code size.

Do you have any suggestions?

Reply:
For case 1 you can use junit `Assume` to skip the param you do not want to test.

Reply:
I have revised them to parameterized tests. Thank you.